contrib/Shopify/sarama/option.go

```diff
@@ -7,15 +7,22 @@ package sarama
 
 import (
 	"math"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
 )
 
 type config struct {
-	serviceName   string
-	analyticsRate float64
+	consumerServiceName string
+	producerServiceName string
+	analyticsRate       float64
 }
 
 func defaults(cfg *config) {
-	cfg.serviceName = "kafka"
+	cfg.consumerServiceName = "kafka"
+	cfg.producerServiceName = globalconfig.ServiceName()
+	if cfg.producerServiceName == "" {
```

Comment:
Seems a bit backwards to me. Can we try this instead:
```go
cfg.consumerServiceName = "kafka"
cfg.producerServiceName = "kafka"
if svc := globalconfig.ServiceName(); svc != "" {
    cfg.producerServiceName = svc
}
```
What do you think? Better? Worse? Meah?

Reply:
👍 Responded below.